src/worker.js

```diff
@@ -54,16 +39,19 @@ create().onRequest('job', ({ contents, type, config, filePath, projectPath, rule
   const configPath = Helpers.getConfigPath(fileDir)
   const relativeFilePath = Helpers.getRelativePath(fileDir, filePath, config)
 
-  const argv = Helpers.getArgv(type, config, rules, relativeFilePath, fileDir, configPath)
+  const cliEngineOptions =
+    Helpers.getCLIEngineOptions(type, config, rules, relativeFilePath, fileDir, configPath)
 
-  if (type === 'lint') {
-    job.response = lintJob(argv, contents, eslint, configPath, config)
+  const noProjectConfig = (configPath === null || isConfigAtHomeRoot(configPath))
+  if (noProjectConfig && config.disableWhenNoEslintConfig) {
+    job.response = []
+  } else if (type === 'lint') {
+    const report = lintJob({ cliEngineOptions, contents, eslint, filePath })
+    job.response = report.results.length ? report.results[0].messages : []
   } else if (type === 'fix') {
-    job.response = fixJob(argv, eslint)
+    job.response = fixJob({ cliEngineOptions, contents, eslint, filePath })
```

Comment:
Question: Is the reason you're passing a currently-always-undefined `contents` into `fixJob` because you expect that in the future we would pass unsaved text into the fix job?  For some reason I thought that ESLint wouldn't allow `--fix` for stdin, but maybe it does when using `executeOnText`?  Could you talk a little about your thoughts here on passing `contents` to `fixJob`?